src/main/java/com/example/chapter1/basicprogramming/BinarySearchRecursively.java

```diff
@@ -0,0 +1,25 @@
+package com.example.chapter1.basicprogramming;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.util.Arrays;
+
+public class BinarySearchRecursively {
+
+
+    public static int rank(int[] input, int searchElement) {
+        if (ArrayUtils.isEmpty(input)) throw new IllegalArgumentException("Empty array.");
```

Comment:
For now we will go with the rule: `Static methods should get accessed with its class name.`